controller/update.go

```diff
@@ -10,6 +10,22 @@ import (
 	"github.com/giantswarm/inago/task"
 )
 
+func (c controller) executeTaskAction(f func(req Request) (*task.Task, error), req Request) error {
+	taskObject, err := f(req)
+	if err != nil {
+		return maskAny(err)
+	}
+	closer := make(<-chan struct{})
+	taskObject, err = c.WaitForTask(taskObject.ID, closer)
+	if err != nil {
+		return maskAny(err)
+	}
+	if task.HasFailedStatus(taskObject) {
+		return maskAny(fmt.Errorf(taskObject.Error))
```

Comment:
this is the actual fix, which we didn't do everywhere. See first commit. I then noticed the code duplication and cleaned that up too. Thus the title is a little bit misleading.